docs/api/modules/Internationalization.md

```diff
@@ -0,0 +1,196 @@
+[👈 Return to Overview](../API.md)
+
+# Internationalization
+
+To enable internationalization modules can contain a locale directory in the root of
+the module. Messages for each locale should be stored in JSON files within
+the locale directory with filenames using the BCP-47 formatted language tag
+(e.g. en-US.json).
+
+**Contents**
+* [Locale Structure](#locale-structure)
+  - * [Environment Specific Data](#environment-specific-data)
+  - * [ONE_CLIENT_LOCALE_FILENAME](#`ONE_CLIENT_LOCALE_FILENAME`)
+* [Loading Language Packs](#loading-language-packs)
+  - * [`loadLanguagePack`](#`loadLanguagePack`)
+  - * [`queryLanguagePack`](#`queryLanguagePack`)
+  - * [`updateLocale`](#`updateLocale`)
+  - * [`getLocalePack`](#`getLocalePack`)
+
+
+## Locale Structure
+
+Following the below structure will allow [one-app-locale-bundler](https://github.com/americanexpress/one-app-cli/tree/master/packages/one-app-locale-bundler) to correctly
+bundle your locale files.
+
+Module Root:
+```
+module
+└── locale
+| └── en-US.json
+| └── es-MX.json
+|── src
+|   └── index.js
+└── package.json
+```
+
+This will result in:
+```
+build
+└── 1.0.0
+    ├── <moduleName>.browser.js
+    ├── <moduleName>.legacy.browser.js
+    ├── <moduleName>.node.js
+    └── en-us
+    |    ├── <moduleName>.json
+    |    ├── integration.json
+    |    └── qa.json
+    └──es-MX
+        ├── <moduleName>.json
+        ├── integration.json
+        └── qa.json
+```
+
+The language pack itself needs to be a flat JSON object and will not work with a nested structure.
+
+Example `en-US.json`:
+```JSON
+{
+  "loading": "Loading...",
+  "welcome": "welcome to my module"
+}
+```
+
+### Environment Specific Data
+
+[one-app-locale-bundler](https://github.com/americanexpress/one-app-cli/tree/master/packages/one-app-locale-bundler) also allows your module to provide environment
```

Comment:
If dash syntax for a library is used I would add backticks to it.

Reply:
We should standardize on formatting for this. I know I often avoid using backticks in links bc it renders oddly. @mtomcal you had a styleguide right?

Reply:
Idk that we want to call out `one-app-locale-bundler` here since most users will not use it directly, but through `one-app-bundler`

Reply:
Yes, I need to create a "Do's and Don'ts" section for some of the organic decisions we made this past week on style. I think backticks with links is important. When the docsite is built, we will need to make sure these links look good.